server/src/auth/local.es6

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+/**
+ * Module for local stub authentication in development and test. This
+ * middleware adds a user to every request, skipping any authentication checks.
+ * @module auth/local
+ */
+
+const util = require('../services/util.es6');
+
+function localAuth(req, res, next) {
+  // Test authentication is enabled e.g do nothing and use production
+  // authentication
+  if (!util.isLocalOrCI()) {
```

Comment:
I ended up looking more broadly in where this gets used. and it looks like that it's only used a couple of other places now - like setting the AuthEmail in the special use controllers. I wonder if we might want to just add that in the middlelayer here as its kind of related to who the user is/ and their ability to access resources.

Reply:
Also since we're now only calilng this middlelayer if is not in production - do we need to have this conditional?

Reply:
Right! I have one more [follow-up PR](https://github.com/18F/fs-open-forest-platform/compare/346-3-test-database) that I'm almost done with that will remove the remaining occurrences of `localOrCi()`. I missed AuthEmail in this one, but made the change in the following. Basically `localOrCI()` becomes `!util.isProduction()`.

Reply:
The only thing about `!util.isProduction()` is that we currently have authentication in our staging environments.

Reply:
It's a bit confusing but production here is really referring to the configuration, not the environment. So the staging environment is also `process.env.NODE_ENV === 'production'` and will pass this check. So it will also have authentication.